subprojects/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/ConfigurationCacheJavaIntegrationTest.groovy

```diff
@@ -30,6 +30,64 @@ class ConfigurationCacheJavaIntegrationTest extends AbstractConfigurationCacheIn
         configurationCache = newConfigurationCacheFixture()
     }
 
+    def "can add JavaExec mainClass convention to automatically find class at execution time"() {
```

Comment:
I think a better home for this might be the tests for the `JavaExec` task, as this use case should work regardless of whether the configuration cache is being used or not.

Reply:
Agreed. Done in https://github.com/gradle/gradle/pull/14351. I kept the configuration cache assertions though, what do you think?